teuthology/misc.py

```diff
@@ -764,7 +764,15 @@ def pull_directory(remote, remotedir, localdir):
     if not os.path.exists(localdir):
         os.mkdir(localdir)
     r = remote.get_tar_stream(remotedir, sudo=True)
-    tar = tarfile.open(mode='r|gz', fileobj=r.stdout)
+    try:
+        tar = tarfile.open(mode='r|gz', fileobj=r.stdout)
```

Comment:
I just wonder why this check should be here? Why not to handle this by caller of `pull_directory`?
The caller should be aware of existence of a remote directory, if it unsure then it should handle the call correctly.
From other side, can the description of the function be updated to reflect the behavior what caller should expect from the function.